RecoMET/METFilters/interface/EcalBoundaryInfoCalculator.h

```diff
@@ -272,32 +272,24 @@ template<class EcalDetId> class EcalBoundaryInfoCalculator {
         return nextDirection;
     }
 
-    void initializeEcalNavigator(DetId startE, const edm::ESHandle<CaloTopology> theCaloTopology,
-                                 EcalSubdetector ecalSubDet) {
+    CaloNavigator<EcalDetId> * initializeEcalNavigator(DetId startE, const edm::ESHandle<CaloTopology> theCaloTopology,

```

Comment:
How about changing this to return a `std::unique_ptr<CaloNavigator<EcalDetId>`. That would simplify the logic in the routine which uses it.